style-dictionary.config.js

```diff
@@ -89,14 +89,38 @@ function minifyCSSVarDictionary(obj) {
   return toRet;
 }
 
+/**
+ * Tokens with key '@' mean the value of the parent is the value of '@', e.g.:
+ * {background: {neutral: {@: 'value' }}} is compiled to `background-neutral-default-@: 'value'`
+ * but we want this to look like `background-neutral: 'value'`.
```

Comment:
This first block is the only part of this PR I think could be better. After seeing the change happening in `src/tokens-dist/json/css-variables-nested.json`, the `formatEdsTokens` code was pretty easy to understand. Nested loops are always a little confusing, but I think this one's fine. So my suggestion is to put some of that json from the changes in `src/tokens-dist/json/css-variables-nested.json` into the comment since I found that very helpful. This makes it a pretty long comment though, so I understand if you want to leave the example out or make it more compact. And my version could probably also be a lot easier to understand, but it's my best idea at the moment.

```suggestion
 * Tokens with key '@' mean the value of the parent is the value of '@', e.g.:
 * {background: {neutral: {@: 'value' }}} is compiled to `background-neutral-default-@: 'value'`
 * but we want this to look like `background-neutral: 'value'`.
```
```suggestion
 * Tokens with key '@' are the base value of the parent, e.g.
 * {background: {neutral: {@: 'value' }}} is compiled to `background-neutral-default-@: 'value'`,
 * but we want this to look like `background-neutral: 'value'`.
 * 
 * This function moves the '@' token out to be a sibling of the parent with the '@' part of
 * the name removed.
 *
 * Example:
 * "neutral": {
 *   "default": {
 *     "@": "var(--eds-theme-color-border-neutral-default)",
 *     "hover": "var(--eds-theme-color-border-neutral-default-hover)"
 *   },
 * },
 *
 * will be changed to
 * 
 * "neutral": {
 *   "default": {
 *     "hover": "var(--eds-theme-color-border-neutral-default-hover)"
 *   },
 * },
 * "neutral-default": "var(--eds-theme-color-border-neutral-default)",
```

Reply:
I'm down for adding the example for clarification, as there is value in showing what was happening and what we want out of the helper function